src/main/java/nextstep/optional/User.java

```diff
@@ -33,7 +35,13 @@ public static boolean ageIsInRange1(User user) {
     }
 
     public static boolean ageIsInRange2(User user) {
-        return false;
+        return Optional.ofNullable(user)
+                       .stream()
+                       .map(User::getAge)
+                       .map(Optional::ofNullable)
+                       .filter(Optional::isPresent)
+                       .map(Optional::get)
+                       .anyMatch(age -> age >= 30 && age <= 45);
```

Comment:
`Stream` 을 사용할 필요가 있을까요? 
연산이 꽤 많아 보이네요. 

```suggestion
        return Optional.ofNullable(user)
            .map(it -> it.age)
            .filter(it -> it >= 30 && it <= 45)
            .isPresent();
```

Reply:
반영하겠습니다~